<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>curatedMetagenomicData • curatedMetagenomicData</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="curatedMetagenomicData">
<meta property="og:description" content="curatedMetagenomicData">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">curatedMetagenomicData</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">3.0.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/curatedMetagenomicData.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/articles/available-studies.html">Available Studies</a>
    </li>
    <li>
      <a href="../articles/articles/our-pipeline.html">Our Pipeline</a>
    </li>
    <li>
      <a href="../articles/articles/version-three.html">Version Three</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/waldronlab/curatedMetagenomicData/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="curatedMetagenomicData_files/header-attrs-2.9/header-attrs.js"></script><script src="curatedMetagenomicData_files/htmlwidgets-1.5.3/htmlwidgets.js"></script><script src="curatedMetagenomicData_files/jquery-3.5.1/jquery.min.js"></script><link href="curatedMetagenomicData_files/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="curatedMetagenomicData_files/datatables-binding-0.18/datatables.js"></script><link href="curatedMetagenomicData_files/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="curatedMetagenomicData_files/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="curatedMetagenomicData_files/dt-core-1.10.20/js/jquery.dataTables.min.js"></script><link href="curatedMetagenomicData_files/dt-ext-responsive-1.10.20/css/responsive.dataTables.min.css" rel="stylesheet">
<script src="curatedMetagenomicData_files/dt-ext-responsive-1.10.20/js/dataTables.responsive.min.js"></script><link href="curatedMetagenomicData_files/crosstalk-1.1.1/css/crosstalk.css" rel="stylesheet">
<script src="curatedMetagenomicData_files/crosstalk-1.1.1/js/crosstalk.min.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>curatedMetagenomicData</h1>
                        <h4 class="author">Lucas Schiffer, MPH</h4>
            <address class="author_afil">
      Section of Computational Biomedicine, Boston University School of Medicine, Boston, MA, U.S.A.<br><a class="author_email" href="mailto:#"></a><a href="mailto:schifferl@bu.edu" class="email">schifferl@bu.edu</a>
      </address>
                              <h4 class="author">Levi Waldron, PhD</h4>
            <address class="author_afil">
      Graduate School of Public Health and Health Policy, City University of New York, New York, NY, U.S.A.<br><a class="author_email" href="mailto:#"></a><a href="mailto:levi.waldron@sph.cuny.edu" class="email">levi.waldron@sph.cuny.edu</a>
      </address>
                  
      
      <small class="dont-index">Source: <a href="https://github.com/waldronlab/curatedMetagenomicData/blob/master/vignettes/curatedMetagenomicData.Rmd"><code>vignettes/curatedMetagenomicData.Rmd</code></a></small>
      <div class="hidden name"><code>curatedMetagenomicData.Rmd</code></div>

    </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>The curatedMetagenomicData package provides standardized, curated human microbiome data for novel analyses. It includes gene families, marker abundance, marker presence, pathway abundance, pathway coverage, and relative abundance for samples collected from different body sites. The bacterial, fungal, and archaeal taxonomic abundances for each sample were calculated with MetaPhlAn3 and metabolic functional potential was calculated with HUMAnN3. The manually curated sample metadata and standardized metagenomic data are available as (Tree)SummarizedExperiment objects.</p>
    </div>
    
<div id="installation" class="section level1">
<h1 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h1>
<p>Most users should simply install the current version of <em><a href="https://bioconductor.org/packages/3.13/curatedMetagenomicData">curatedMetagenomicData</a></em> from Bioconductor, unless they have good reason not to. Changes to the package are always commited to GitHub first, and it might be desirable to have the absolute latest changes by installing <em><a href="https://github.com/waldronlab/curatedMetagenomicData">curatedMetagenomicData</a></em> from GitHub; although, this is not recommended for most users. Finally, we have made extensive changes in <em><a href="https://bioconductor.org/packages/3.13/curatedMetagenomicData">curatedMetagenomicData</a></em> <code>3.0.0</code> and some users might want to revert to the previous version of <em><a href="https://github.com/waldronlab/curatedMetagenomicData">curatedMetagenomicData</a></em> from GitHub for older analyses. The previous version of <em><a href="https://github.com/waldronlab/curatedMetagenomicData">curatedMetagenomicData</a></em> from GitHub, <code>1.20.0</code>, will remain installable for the foreseeable future.</p>
<p>To install the current version of <em><a href="https://bioconductor.org/packages/3.13/curatedMetagenomicData">curatedMetagenomicData</a></em> from Bioconductor, use <em><a href="https://CRAN.R-project.org/package=BiocManager">BiocManager</a></em> as follows.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span><span class="op">(</span><span class="st">"curatedMetagenomicData"</span><span class="op">)</span></code></pre></div>
<p>To install the current version of <em><a href="https://github.com/waldronlab/curatedMetagenomicData">curatedMetagenomicData</a></em> from GitHub, use <em><a href="https://CRAN.R-project.org/package=BiocManager">BiocManager</a></em> as follows.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span><span class="op">(</span><span class="st">"waldronlab/curatedMetagenomicData"</span>, dependencies <span class="op">=</span> <span class="cn">TRUE</span>, build_vignettes <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p>To install the previous version of <em><a href="https://github.com/waldronlab/curatedMetagenomicData">curatedMetagenomicData</a></em> from GitHub, use <em><a href="https://CRAN.R-project.org/package=BiocManager">BiocManager</a></em> as follows.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span><span class="op">(</span><span class="st">"waldronlab/curatedMetagenomicData"</span>, dependencies <span class="op">=</span> <span class="cn">TRUE</span>, build_vignettes <span class="op">=</span> <span class="cn">TRUE</span>, ref <span class="op">=</span> <span class="st">"v1.20.0"</span><span class="op">)</span></code></pre></div>
</div>
<div id="r-packages" class="section level1">
<h1 class="hasAnchor">
<a href="#r-packages" class="anchor"></a>R Packages</h1>
<p>To demonstrate the functionality of <em><a href="https://bioconductor.org/packages/3.13/curatedMetagenomicData">curatedMetagenomicData</a></em>, the <em><a href="https://CRAN.R-project.org/package=dplyr">dplyr</a></em> and <em><a href="https://CRAN.R-project.org/package=DT">DT</a></em> packages are needed.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org">dplyr</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/rstudio/DT">DT</a></span><span class="op">)</span></code></pre></div>
</div>
<div id="sample-metadata" class="section level1">
<h1 class="hasAnchor">
<a href="#sample-metadata" class="anchor"></a>Sample Metadata</h1>
<p>The <em><a href="https://bioconductor.org/packages/3.13/curatedMetagenomicData">curatedMetagenomicData</a></em> package contains a <code>data.frame</code>, <code>sampleMetadata</code>, of manually curated sample metadata to help users understand the nature of studies and samples available prior to returning resources. Beyond this, it serves two purposes: 1) to define <code>study_name</code>, which is used with the <code><a href="../reference/curatedMetagenomicData.html">curatedMetagenomicData()</a></code> method to query and return resources, and 2) to define <code>sample_id</code>, which is used with the <code><a href="../reference/returnSamples.html">returnSamples()</a></code> method to return samples across studies.</p>
<p>To demonstrate, the first ten rows and columns (without any <code>NA</code> values) of <code>sampleMetadata</code> for the <code>AsnicarF_2017</code> study are shown in the table below.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sampleMetadata</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">study_name</span> <span class="op">==</span> <span class="st">"AsnicarF_2017"</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="fu">where</span><span class="op">(</span><span class="op">~</span> <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/any.html">any</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">.x</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/slice.html">slice</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://rdrr.io/pkg/DT/man/datatable.html">datatable</a></span><span class="op">(</span>options <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>dom <span class="op">=</span> <span class="st">"t"</span><span class="op">)</span>, extensions <span class="op">=</span> <span class="st">"Responsive"</span><span class="op">)</span></code></pre></div>
<div id="htmlwidget-d413c7421c92a2ecd54e" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-d413c7421c92a2ecd54e">{"x":{"filter":"none","extensions":["Responsive"],"data":[["1","2","3","4","5","6","7","8","9","10"],["AsnicarF_2017","AsnicarF_2017","AsnicarF_2017","AsnicarF_2017","AsnicarF_2017","AsnicarF_2017","AsnicarF_2017","AsnicarF_2017","AsnicarF_2017","AsnicarF_2017"],["MV_FEI1_t1Q14","MV_FEI2_t1Q14","MV_FEI3_t1Q14","MV_FEI4_t1Q14","MV_FEI4_t2Q15","MV_FEI5_t1Q14","MV_FEI5_t2Q14","MV_FEI5_t3Q15","MV_FEM1_t1Q14","MV_FEM2_t1Q14"],["MV_FEI1","MV_FEI2","MV_FEI3","MV_FEI4","MV_FEI4","MV_FEI5","MV_FEI5","MV_FEI5","MV_FEM1","MV_FEM2"],["stool","stool","stool","stool","stool","stool","stool","stool","stool","stool"],["control","control","control","control","control","control","control","control","control","control"],["healthy","healthy","healthy","healthy","healthy","healthy","healthy","healthy","healthy","healthy"],["newborn","newborn","newborn","child","child","child","child","child","adult","adult"],["female","male","male","male","male","male","male","male","female","female"],["ITA","ITA","ITA","ITA","ITA","ITA","ITA","ITA","ITA","ITA"],["no","no","no","no","no","no","no","no","no","no"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>study_name<\/th>\n      <th>sample_id<\/th>\n      <th>subject_id<\/th>\n      <th>body_site<\/th>\n      <th>study_condition<\/th>\n      <th>disease<\/th>\n      <th>age_category<\/th>\n      <th>gender<\/th>\n      <th>country<\/th>\n      <th>non_westernized<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"t","order":[],"autoWidth":false,"orderClasses":false,"columnDefs":[{"orderable":false,"targets":0}],"responsive":true}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="data-access" class="section level1">
<h1 class="hasAnchor">
<a href="#data-access" class="anchor"></a>Data Access</h1>
<p>There are three main ways to access data from curatedMetagenomicData: 1. The <code><a href="../reference/curatedMetagenomicData.html">curatedMetagenomicData()</a></code> function to search and return individual datasets 2. The <code><a href="../reference/returnSamples.html">returnSamples()</a></code> function to access merged datasets based on filtered metadata 3. Through the command-line interface. This is not part of the Bioconductor package or discussed here; see <a href="https://github.com/waldronlab/curatedMetagenomicDataCLI" class="uri">https://github.com/waldronlab/curatedMetagenomicDataCLI</a></p>
<div id="the-curatedmetagenomicdata-function" class="section level2">
<h2 class="hasAnchor">
<a href="#the-curatedmetagenomicdata-function" class="anchor"></a>The curatedMetagenomicData function</h2>
<p>To access curated metagenomic data, users will use the <code><a href="../reference/curatedMetagenomicData.html">curatedMetagenomicData()</a></code> method both to query and return resources. The first argument <code>pattern</code> is a regular expression pattern to look for in the titles of resources available in <em><a href="https://bioconductor.org/packages/3.13/curatedMetagenomicData">curatedMetagenomicData</a></em>; <code>""</code> will return all resources. The title of each resource is a three part string with “.” as a delimiter - the fields are <code>runDate</code>, <code>studyName</code>, and <code>dataType</code>. The <code>runDate</code> is the date we created the resource and can mostly be ignored by users because if there is more than one date corresponding to a resource, the most recent one is selected automatically. It would be used if a specific <code>runDate</code> was needed, but only a single copy of each resource exists as of this writing.</p>
<p>Multiple resources can be queried or returned with a single call to <code><a href="../reference/curatedMetagenomicData.html">curatedMetagenomicData()</a></code>, only the titles of resources are returned by default. The following search term is a regex that matches all six data types from all datasets starting with <code>*AsnicarF_20*</code>:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/curatedMetagenomicData.html">curatedMetagenomicData</a></span><span class="op">(</span><span class="st">"AsnicarF_20.+"</span><span class="op">)</span>
<span class="co">## 2021-03-31.AsnicarF_2017.gene_families</span>
<span class="co">## 2021-03-31.AsnicarF_2017.marker_abundance</span>
<span class="co">## 2021-03-31.AsnicarF_2017.marker_presence</span>
<span class="co">## 2021-03-31.AsnicarF_2017.pathway_abundance</span>
<span class="co">## 2021-03-31.AsnicarF_2017.pathway_coverage</span>
<span class="co">## 2021-03-31.AsnicarF_2017.relative_abundance</span>
<span class="co">## 2021-03-31.AsnicarF_2021.gene_families</span>
<span class="co">## 2021-03-31.AsnicarF_2021.marker_abundance</span>
<span class="co">## 2021-03-31.AsnicarF_2021.marker_presence</span>
<span class="co">## 2021-03-31.AsnicarF_2021.pathway_abundance</span>
<span class="co">## 2021-03-31.AsnicarF_2021.pathway_coverage</span>
<span class="co">## 2021-03-31.AsnicarF_2021.relative_abundance</span></code></pre></div>
<p>If you are more comfortable with wildcard-like searching, the following is equivalent:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/glob2rx.html">glob2rx</a></span><span class="op">(</span><span class="st">"*AsnicarF_20*"</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="../reference/curatedMetagenomicData.html">curatedMetagenomicData</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">## 2021-03-31.AsnicarF_2017.gene_families</span>
<span class="co">## 2021-03-31.AsnicarF_2017.marker_abundance</span>
<span class="co">## 2021-03-31.AsnicarF_2017.marker_presence</span>
<span class="co">## 2021-03-31.AsnicarF_2017.pathway_abundance</span>
<span class="co">## 2021-03-31.AsnicarF_2017.pathway_coverage</span>
<span class="co">## 2021-03-31.AsnicarF_2017.relative_abundance</span>
<span class="co">## 2021-03-31.AsnicarF_2021.gene_families</span>
<span class="co">## 2021-03-31.AsnicarF_2021.marker_abundance</span>
<span class="co">## 2021-03-31.AsnicarF_2021.marker_presence</span>
<span class="co">## 2021-03-31.AsnicarF_2021.pathway_abundance</span>
<span class="co">## 2021-03-31.AsnicarF_2021.pathway_coverage</span>
<span class="co">## 2021-03-31.AsnicarF_2021.relative_abundance</span></code></pre></div>
<p>When the second argument <code>dryrun</code> is set to <code>FALSE</code>, a <code>list</code> of <code>SummarizedExperiment</code> and/or <code>TreeSummarizedExperiment</code> objects is returned. When a single resource is requested, a single element <code>list</code> is returned.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/curatedMetagenomicData.html">curatedMetagenomicData</a></span><span class="op">(</span><span class="st">"AsnicarF_2017.relative_abundance"</span>, dryrun <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="co">## $`2021-03-31.AsnicarF_2017.relative_abundance`</span>
<span class="co">## class: TreeSummarizedExperiment </span>
<span class="co">## dim: 301 24 </span>
<span class="co">## metadata(0):</span>
<span class="co">## assays(1): relative_abundance</span>
<span class="co">## rownames(301):</span>
<span class="co">##   k__Bacteria|p__Proteobacteria|c__Gammaproteobacteria|o__Enterobacterales|f__Enterobacteriaceae|g__Escherichia|s__Escherichia_coli</span>
<span class="co">##   k__Bacteria|p__Actinobacteria|c__Actinobacteria|o__Bifidobacteriales|f__Bifidobacteriaceae|g__Bifidobacterium|s__Bifidobacterium_bifidum</span>
<span class="co">##   ...</span>
<span class="co">##   k__Bacteria|p__Firmicutes|c__Bacilli|o__Lactobacillales|f__Streptococcaceae|g__Streptococcus|s__Streptococcus_gordonii</span>
<span class="co">##   k__Bacteria|p__Firmicutes|c__Bacilli|o__Lactobacillales|f__Aerococcaceae|g__Abiotrophia|s__Abiotrophia_sp_HMSC24B09</span>
<span class="co">## rowData names(7): Kingdom Phylum ... Genus Species</span>
<span class="co">## colnames(24): MV_FEI1_t1Q14 MV_FEI2_t1Q14 ... MV_MIM5_t2M14</span>
<span class="co">##   MV_MIM5_t3F15</span>
<span class="co">## colData names(22): study_name subject_id ... lactating curator</span>
<span class="co">## reducedDimNames(0):</span>
<span class="co">## mainExpName: NULL</span>
<span class="co">## altExpNames(0):</span>
<span class="co">## rowLinks: a LinkDataFrame (301 rows)</span>
<span class="co">## rowTree: 1 phylo tree(s) (10430 leaves)</span>
<span class="co">## colLinks: NULL</span>
<span class="co">## colTree: NULL</span></code></pre></div>
<p>When the third argument <code>counts</code> is set to <code>TRUE</code>, relative abundance proportions are multiplied by read depth and rounded to the nearest integer prior to being returned. Also, when multiple resources are requested, the <code>list</code> will contain named elements corresponding to each <code>SummarizedExperiment</code> and/or <code>TreeSummarizedExperiment</code> object.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/curatedMetagenomicData.html">curatedMetagenomicData</a></span><span class="op">(</span><span class="st">"AsnicarF_20.+.relative_abundance"</span>, dryrun <span class="op">=</span> <span class="cn">FALSE</span>, counts <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="co">## $`2021-03-31.AsnicarF_2017.relative_abundance`</span>
<span class="co">## class: TreeSummarizedExperiment </span>
<span class="co">## dim: 301 24 </span>
<span class="co">## metadata(0):</span>
<span class="co">## assays(1): relative_abundance</span>
<span class="co">## rownames(301):</span>
<span class="co">##   k__Bacteria|p__Proteobacteria|c__Gammaproteobacteria|o__Enterobacterales|f__Enterobacteriaceae|g__Escherichia|s__Escherichia_coli</span>
<span class="co">##   k__Bacteria|p__Actinobacteria|c__Actinobacteria|o__Bifidobacteriales|f__Bifidobacteriaceae|g__Bifidobacterium|s__Bifidobacterium_bifidum</span>
<span class="co">##   ...</span>
<span class="co">##   k__Bacteria|p__Firmicutes|c__Bacilli|o__Lactobacillales|f__Streptococcaceae|g__Streptococcus|s__Streptococcus_gordonii</span>
<span class="co">##   k__Bacteria|p__Firmicutes|c__Bacilli|o__Lactobacillales|f__Aerococcaceae|g__Abiotrophia|s__Abiotrophia_sp_HMSC24B09</span>
<span class="co">## rowData names(7): Kingdom Phylum ... Genus Species</span>
<span class="co">## colnames(24): MV_FEI1_t1Q14 MV_FEI2_t1Q14 ... MV_MIM5_t2M14</span>
<span class="co">##   MV_MIM5_t3F15</span>
<span class="co">## colData names(22): study_name subject_id ... lactating curator</span>
<span class="co">## reducedDimNames(0):</span>
<span class="co">## mainExpName: NULL</span>
<span class="co">## altExpNames(0):</span>
<span class="co">## rowLinks: a LinkDataFrame (301 rows)</span>
<span class="co">## rowTree: 1 phylo tree(s) (10430 leaves)</span>
<span class="co">## colLinks: NULL</span>
<span class="co">## colTree: NULL</span>
<span class="co">## </span>
<span class="co">## $`2021-03-31.AsnicarF_2021.relative_abundance`</span>
<span class="co">## class: TreeSummarizedExperiment </span>
<span class="co">## dim: 639 1098 </span>
<span class="co">## metadata(0):</span>
<span class="co">## assays(1): relative_abundance</span>
<span class="co">## rownames(639):</span>
<span class="co">##   k__Bacteria|p__Bacteroidetes|c__Bacteroidia|o__Bacteroidales|f__Bacteroidaceae|g__Bacteroides|s__Bacteroides_vulgatus</span>
<span class="co">##   k__Bacteria|p__Bacteroidetes|c__Bacteroidia|o__Bacteroidales|f__Bacteroidaceae|g__Bacteroides|s__Bacteroides_stercoris</span>
<span class="co">##   ...</span>
<span class="co">##   k__Bacteria|p__Synergistetes|c__Synergistia|o__Synergistales|f__Synergistaceae|g__Pyramidobacter|s__Pyramidobacter_sp_C12_8</span>
<span class="co">##   k__Bacteria|p__Actinobacteria|c__Actinobacteria|o__Micrococcales|f__Brevibacteriaceae|g__Brevibacterium|s__Brevibacterium_aurantiacum</span>
<span class="co">## rowData names(7): Kingdom Phylum ... Genus Species</span>
<span class="co">## colnames(1098): SAMEA7041133 SAMEA7041134 ... SAMEA7045952 SAMEA7045953</span>
<span class="co">## colData names(21): study_name subject_id ... curator BMI</span>
<span class="co">## reducedDimNames(0):</span>
<span class="co">## mainExpName: NULL</span>
<span class="co">## altExpNames(0):</span>
<span class="co">## rowLinks: a LinkDataFrame (639 rows)</span>
<span class="co">## rowTree: 1 phylo tree(s) (10430 leaves)</span>
<span class="co">## colLinks: NULL</span>
<span class="co">## colTree: NULL</span></code></pre></div>
<p>If you are interested only in taxonomic relative abundance, include <code>*relative_abundance*</code> in your search term, like above or with searches like <code>glob2rx(*AsnicarF_20*abundance)</code>.</p>
<div id="merging-curatedmetagenomicdata-results" class="section level3">
<h3 class="hasAnchor">
<a href="#merging-curatedmetagenomicdata-results" class="anchor"></a>Merging <code>curatedMetagenomicData()</code> results</h3>
<p>To merge the <code>list</code> elements returned from the <code><a href="../reference/curatedMetagenomicData.html">curatedMetagenomicData()</a></code> method into a single <code>SummarizedExperiment</code> or <code>TreeSummarizedExperiment</code> object, users will use the <code><a href="../reference/mergeData.html">mergeData()</a></code> method, provided elements are the same <code>dataType</code>.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/curatedMetagenomicData.html">curatedMetagenomicData</a></span><span class="op">(</span><span class="st">"LiJ_20.+.marker_abundance"</span>, dryrun <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="../reference/mergeData.html">mergeData</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">## class: SummarizedExperiment </span>
<span class="co">## dim: 77712 456 </span>
<span class="co">## metadata(0):</span>
<span class="co">## assays(1): marker_abundance</span>
<span class="co">## rownames(77712): 39491__A0A395UVM9__DXB76_04540</span>
<span class="co">##   39491__A0A395UZL1__DXB76_05950 ... 1262937__R7LHU6__BN805_01557</span>
<span class="co">##   712117__F3PBR4__HMPREF9056_02502</span>
<span class="co">## rowData names(0):</span>
<span class="co">## colnames(456): DLF005-IE DLM006-IE ... nHM612836 nHMX11726</span>
<span class="co">## colData names(28): study_name subject_id ... location smoker</span></code></pre></div>
<p>The <code><a href="../reference/mergeData.html">mergeData()</a></code> method works for every <code>dataType</code> and will always return the appropriate data structure (a single <code>SummarizedExperiment</code> or <code>TreeSummarizedExperiment</code> object).</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/curatedMetagenomicData.html">curatedMetagenomicData</a></span><span class="op">(</span><span class="st">"LiJ_20.+.pathway_abundance"</span>, dryrun <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="../reference/mergeData.html">mergeData</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">## class: SummarizedExperiment </span>
<span class="co">## dim: 27110 456 </span>
<span class="co">## metadata(0):</span>
<span class="co">## assays(1): pathway_abundance</span>
<span class="co">## rownames(27110): UNMAPPED UNINTEGRATED ... PWY-7539:</span>
<span class="co">##   6-hydroxymethyl-dihydropterin diphosphate biosynthesis III</span>
<span class="co">##   (Chlamydia)|g__Prevotella.s__Prevotella_buccae LACTOSECAT-PWY:</span>
<span class="co">##   lactose and galactose degradation</span>
<span class="co">##   I|g__Escherichia.s__Escherichia_coli</span>
<span class="co">## rowData names(0):</span>
<span class="co">## colnames(456): DLF005-IE DLM006-IE ... nHM612836 nHMX11726</span>
<span class="co">## colData names(28): study_name subject_id ... location smoker</span></code></pre></div>
<p>This is useful for analysis across entire studies (e.g. meta-analysis); however, when doing analysis across individual samples (e.g. mega-analysis) the <code><a href="../reference/returnSamples.html">returnSamples()</a></code> method is preferable.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/curatedMetagenomicData.html">curatedMetagenomicData</a></span><span class="op">(</span><span class="st">"LiJ_20.+.relative_abundance"</span>, dryrun <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="../reference/mergeData.html">mergeData</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">## class: TreeSummarizedExperiment </span>
<span class="co">## dim: 688 456 </span>
<span class="co">## metadata(0):</span>
<span class="co">## assays(1): relative_abundance</span>
<span class="co">## rownames(688):</span>
<span class="co">##   k__Bacteria|p__Bacteroidetes|c__Bacteroidia|o__Bacteroidales|f__Bacteroidaceae|g__Bacteroides|s__Bacteroides_plebeius</span>
<span class="co">##   k__Bacteria|p__Bacteroidetes|c__Bacteroidia|o__Bacteroidales|f__Bacteroidaceae|g__Bacteroides|s__Bacteroides_caccae</span>
<span class="co">##   ...</span>
<span class="co">##   k__Bacteria|p__Proteobacteria|c__Betaproteobacteria|o__Neisseriales|f__Neisseriaceae|g__Neisseria|s__Neisseria_elongata</span>
<span class="co">##   k__Bacteria|p__Proteobacteria|c__Gammaproteobacteria|o__Enterobacterales|f__Morganellaceae|g__Providencia|s__Providencia_alcalifaciens</span>
<span class="co">## rowData names(7): Kingdom Phylum ... Genus Species</span>
<span class="co">## colnames(456): DLF005-IE DLM006-IE ... nHM612836 nHMX11726</span>
<span class="co">## colData names(28): study_name subject_id ... location smoker</span>
<span class="co">## reducedDimNames(0):</span>
<span class="co">## mainExpName: NULL</span>
<span class="co">## altExpNames(0):</span>
<span class="co">## rowLinks: a LinkDataFrame (688 rows)</span>
<span class="co">## rowTree: 1 phylo tree(s) (10430 leaves)</span>
<span class="co">## colLinks: NULL</span>
<span class="co">## colTree: NULL</span></code></pre></div>
</div>
</div>
<div id="the-returnsamples-function-for-merged-datasets-based-on-filtered-metadata" class="section level2">
<h2 class="hasAnchor">
<a href="#the-returnsamples-function-for-merged-datasets-based-on-filtered-metadata" class="anchor"></a>The <code>returnSamples()</code> function for merged datasets based on filtered metadata</h2>
<p>The <code><a href="../reference/returnSamples.html">returnSamples()</a></code> takes a subset of the <code>sampleMetadata</code> object as input, and returns a single <code>SummarizedExperiment</code> or <code>TreeSummarizedExperiment</code> containing one of <em><a href="https://bioconductor.org/packages/3.13/curatedMetagenomicData">curatedMetagenomicData</a></em>’s data types, with this metadata subset as its <code>colData</code>. To use this method, filter/select rows and columns of interest from the <code>sampleMetadata</code> object, maintaining at least one row, and the sampleID and study_name columns. Provide the resulting <code>data.frame</code> as the first argument to the <code><a href="../reference/returnSamples.html">returnSamples()</a></code> function.</p>
<p>The <code><a href="../reference/returnSamples.html">returnSamples()</a></code> method requires a second argument <code>dataType</code> (<code>"gene_families"</code>, <code>"marker_abundance"</code>, <code>"marker_presence"</code>, <code>"pathway_abundance"</code>, <code>"pathway_coverage"</code>, or <code>"relative_abundance"</code>) be specified. It is often most convenient to subset the <code>sampleMetadata</code> <code>data.frame</code> using <em><a href="https://CRAN.R-project.org/package=dplyr">dplyr</a></em> syntax.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sampleMetadata</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">age</span> <span class="op">&gt;=</span> <span class="fl">18</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">alcohol</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">body_site</span> <span class="op">==</span> <span class="st">"stool"</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="fu">where</span><span class="op">(</span><span class="op">~</span> <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/all.html">all</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">.x</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="../reference/returnSamples.html">returnSamples</a></span><span class="op">(</span><span class="st">"relative_abundance"</span><span class="op">)</span>
<span class="co">## class: TreeSummarizedExperiment </span>
<span class="co">## dim: 833 702 </span>
<span class="co">## metadata(0):</span>
<span class="co">## assays(1): relative_abundance</span>
<span class="co">## rownames(833):</span>
<span class="co">##   k__Bacteria|p__Bacteroidetes|c__Bacteroidia|o__Bacteroidales|f__Prevotellaceae|g__Prevotella|s__Prevotella_copri</span>
<span class="co">##   k__Bacteria|p__Bacteroidetes|c__Bacteroidia|o__Bacteroidales|f__Prevotellaceae|g__Prevotella|s__Prevotella_sp_CAG_520</span>
<span class="co">##   ...</span>
<span class="co">##   k__Bacteria|p__Actinobacteria|c__Actinobacteria|o__Corynebacteriales|f__Corynebacteriaceae|g__Corynebacterium|s__Corynebacterium_aurimucosum</span>
<span class="co">##   k__Bacteria|p__Actinobacteria|c__Actinobacteria|o__Corynebacteriales|f__Corynebacteriaceae|g__Corynebacterium|s__Corynebacterium_coyleae</span>
<span class="co">## rowData names(7): Kingdom Phylum ... Genus Species</span>
<span class="co">## colnames(702): JAS_1 JAS_10 ... YSZC12003_37879 YSZC12003_37880</span>
<span class="co">## colData names(46): study_name subject_id ...</span>
<span class="co">##   age_twins_started_to_live_apart zigosity</span>
<span class="co">## reducedDimNames(0):</span>
<span class="co">## mainExpName: NULL</span>
<span class="co">## altExpNames(0):</span>
<span class="co">## rowLinks: a LinkDataFrame (833 rows)</span>
<span class="co">## rowTree: 1 phylo tree(s) (10430 leaves)</span>
<span class="co">## colLinks: NULL</span>
<span class="co">## colTree: NULL</span></code></pre></div>
<p>The <code>counts</code> argument applies to the <code><a href="../reference/returnSamples.html">returnSamples()</a></code> method as well, and can be passed as the third argument. Finally, users should know that any arbitrary columns added to <code>sampleMetadata</code> will be present in the <code>colData</code> of the <code>SummarizedExperiment</code> or <code>TreeSummarizedExperiment</code> object that is returned.</p>
</div>
</div>
<div id="example-analysis" class="section level1">
<h1 class="hasAnchor">
<a href="#example-analysis" class="anchor"></a>Example Analysis</h1>
<p>To demonstrate the utility of <em><a href="https://bioconductor.org/packages/3.13/curatedMetagenomicData">curatedMetagenomicData</a></em>, an example analysis is presented below. However, readers should know analysis is generally beyond the scope of <em><a href="https://bioconductor.org/packages/3.13/curatedMetagenomicData">curatedMetagenomicData</a></em> and the analysis presented here is for demonstration alone. It is best to consider the output of <em><a href="https://bioconductor.org/packages/3.13/curatedMetagenomicData">curatedMetagenomicData</a></em> as the input of analysis more than anything else.</p>
<div id="r-packages-1" class="section level2">
<h2 class="hasAnchor">
<a href="#r-packages-1" class="anchor"></a>R Packages</h2>
<p>To demonstrate the utility of <em><a href="https://bioconductor.org/packages/3.13/curatedMetagenomicData">curatedMetagenomicData</a></em>, the <em><a href="https://CRAN.R-project.org/package=stringr">stringr</a></em>, <em><a href="https://bioconductor.org/packages/3.13/mia">mia</a></em>, <em><a href="https://bioconductor.org/packages/3.13/scater">scater</a></em>, and <em><a href="https://CRAN.R-project.org/package=vegan">vegan</a></em> packages are needed.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://stringr.tidyverse.org">stringr</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/microbiome/mia">mia</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://bioconductor.org/packages/scater/">scater</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://cran.r-project.org">vegan</a></span><span class="op">)</span></code></pre></div>
</div>
<div id="prepare-data" class="section level2">
<h2 class="hasAnchor">
<a href="#prepare-data" class="anchor"></a>Prepare Data</h2>
<p>In our hypothetical study, let’s examine the association of alcohol consumption and stool microbial composition across all annotated samples in <em><a href="https://bioconductor.org/packages/3.13/curatedMetagenomicData">curatedMetagenomicData</a></em>. We will examine the alpha diversity (within subject diversity), beta diversity (between subject diversity), and conclude with a few notes on differential abundance analysis.</p>
<div id="return-samples" class="section level3">
<h3 class="hasAnchor">
<a href="#return-samples" class="anchor"></a>Return Samples</h3>
<p>First, as above, we use the <code><a href="../reference/returnSamples.html">returnSamples()</a></code> method to return the relevant samples across all studies available in <em><a href="https://bioconductor.org/packages/3.13/curatedMetagenomicData">curatedMetagenomicData</a></em>. We want adults over the age of 18, for whom alcohol consumption status is known, and we want only stool samples. The <code>select(where...</code> line below simply removes metadata columns which are all <code>NA</code> values - they exist in another study but are all <code>NA</code> once subsetting has been done. Lastly, the <code>"relative_abundance"</code> <code>dataType</code> is requested because it contains the relevant information about microbial composition.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">alcoholStudy</span> <span class="op">&lt;-</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">sampleMetadata</span>, <span class="va">age</span> <span class="op">&gt;=</span> <span class="fl">18</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">alcohol</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">body_site</span> <span class="op">==</span> <span class="st">"stool"</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="fu">where</span><span class="op">(</span><span class="op">~</span> <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/all.html">all</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">.x</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="../reference/returnSamples.html">returnSamples</a></span><span class="op">(</span><span class="st">"relative_abundance"</span><span class="op">)</span></code></pre></div>
</div>
<div id="mutate-coldata" class="section level3">
<h3 class="hasAnchor">
<a href="#mutate-coldata" class="anchor"></a>Mutate colData</h3>
<p>Most of the values in the <code>sampleMetadata</code> <code>data.frame</code> (which becomes <code>colData</code>) are in lower snake case (e.g. <code>snake_case</code>) and don’t look nice in plots. Here, the values of the <code>alcohol</code> variable are made into title case using <em><a href="https://CRAN.R-project.org/package=stringr">stringr</a></em> so they will look nice in plots.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">colData</span><span class="op">(</span><span class="va">alcoholStudy</span><span class="op">)</span> <span class="op">&lt;-</span>
    <span class="fu">colData</span><span class="op">(</span><span class="va">alcoholStudy</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>alcohol <span class="op">=</span> <span class="fu"><a href="https://stringr.tidyverse.org/reference/str_replace.html">str_replace_all</a></span><span class="op">(</span><span class="va">alcohol</span>, <span class="st">"no"</span>, <span class="st">"No"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>alcohol <span class="op">=</span> <span class="fu"><a href="https://stringr.tidyverse.org/reference/str_replace.html">str_replace_all</a></span><span class="op">(</span><span class="va">alcohol</span>, <span class="st">"yes"</span>, <span class="st">"Yes"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu">DataFrame</span><span class="op">(</span><span class="op">)</span></code></pre></div>
</div>
<div id="agglomerate-ranks" class="section level3">
<h3 class="hasAnchor">
<a href="#agglomerate-ranks" class="anchor"></a>Agglomerate Ranks</h3>
<p>Next, the <code>splitByRanks</code> method from <em><a href="https://bioconductor.org/packages/3.13/mia">mia</a></em> is used to create alternative experiments for each level of the taxonomic tree (e.g. Genus). This allows for diversity and differential abundance analysis at specific taxonomic levels; with this step complete, our data is ready to analyze.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">altExps</span><span class="op">(</span><span class="va">alcoholStudy</span><span class="op">)</span> <span class="op">&lt;-</span>
    <span class="fu"><a href="https://rdrr.io/pkg/mia/man/splitByRanks.html">splitByRanks</a></span><span class="op">(</span><span class="va">alcoholStudy</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div id="alpha-diversity" class="section level2">
<h2 class="hasAnchor">
<a href="#alpha-diversity" class="anchor"></a>Alpha Diversity</h2>
<p>Alpha diversity is a measure of the within sample diversity of features (relative abundance proportions here) and seeks to quantify the evenness (i.e. are the amounts of different microbes the same) and richness (i.e. are they are large variety of microbial taxa present). The Shannon index (H’) is a commonly used measure of alpha diversity, it’s estimated here using the <code><a href="https://rdrr.io/pkg/mia/man/estimateDiversity.html">estimateDiversity()</a></code> method from the <em><a href="https://bioconductor.org/packages/3.13/mia">mia</a></em> package.</p>
<p>To quickly plot the results of alpha diversity estimation, the <code><a href="https://rdrr.io/pkg/scater/man/plotColData.html">plotColData()</a></code> method from the <em><a href="https://bioconductor.org/packages/3.13/scater">scater</a></em> package is used along with <em><a href="https://CRAN.R-project.org/package=ggplot2">ggplot2</a></em> syntax.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">alcoholStudy</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://rdrr.io/pkg/mia/man/estimateDiversity.html">estimateDiversity</a></span><span class="op">(</span>abund_values <span class="op">=</span> <span class="st">"relative_abundance"</span>, index <span class="op">=</span> <span class="st">"shannon"</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://rdrr.io/pkg/scater/man/plotColData.html">plotColData</a></span><span class="op">(</span>x <span class="op">=</span> <span class="st">"alcohol"</span>, y <span class="op">=</span> <span class="st">"shannon"</span>, colour_by <span class="op">=</span> <span class="st">"alcohol"</span>, shape_by <span class="op">=</span> <span class="st">"alcohol"</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="st">"Alcohol"</span>, y <span class="op">=</span> <span class="st">"Alpha Diversity (H')"</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu">guides</span><span class="op">(</span>color <span class="op">=</span> <span class="fu">guide_none</span><span class="op">(</span><span class="op">)</span>, shape <span class="op">=</span> <span class="fu">guide_none</span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu">theme</span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span></code></pre></div>
<div class="figure">
<img src="curatedMetagenomicData_files/figure-html/unnamed-chunk-19-1.png" alt="Alpha Diversity - Shannon Index (H')" width="700"><p class="caption">
Alpha Diversity - Shannon Index (H’)
</p>
</div>
<p>The figure suggest that those who consume alcohol have higher Shannon alpha diversity than those who do not consume alcohol; however, the difference does not appear to be significant, at least qualitatively.</p>
</div>
<div id="beta-diversity" class="section level2">
<h2 class="hasAnchor">
<a href="#beta-diversity" class="anchor"></a>Beta Diversity</h2>
<p>Beta diversity is a measure of the between sample diversity of features (relative abundance proportions here) and seeks to quantify the magnitude of differences (or similarity) between every given pair of samples. Below it is accessed by Bray–Curtis Principal Coordinates Analysis (PCoA), which is a linear method, and Uniform Manifold Approximation and Projection (UMAP), which is a non-linear method.</p>
<div id="braycurtis-pcoa" class="section level3">
<h3 class="hasAnchor">
<a href="#braycurtis-pcoa" class="anchor"></a>Bray–Curtis PCoA</h3>
<p>To calculate pairwise Bray–Curtis distance for every sample in our study we will use the <code><a href="https://rdrr.io/pkg/scater/man/runMDS.html">runMDS()</a></code> method from the <em><a href="https://bioconductor.org/packages/3.13/scater">scater</a></em> package along with the <code><a href="https://rdrr.io/pkg/vegan/man/vegdist.html">vegdist()</a></code> method from the <em><a href="https://CRAN.R-project.org/package=vegan">vegan</a></em> package.</p>
<p>To quickly plot the results of beta diversity analysis, the <code><a href="https://rdrr.io/pkg/scater/man/plotReducedDim.html">plotReducedDim()</a></code> method from the <em><a href="https://bioconductor.org/packages/3.13/scater">scater</a></em> package is used along with <em><a href="https://CRAN.R-project.org/package=ggplot2">ggplot2</a></em> syntax.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">alcoholStudy</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://rdrr.io/pkg/scater/man/runMDS.html">runMDS</a></span><span class="op">(</span>FUN <span class="op">=</span> <span class="va">vegdist</span>, method <span class="op">=</span> <span class="st">"bray"</span>, exprs_values <span class="op">=</span> <span class="st">"relative_abundance"</span>, altexp <span class="op">=</span> <span class="st">"Genus"</span>, name <span class="op">=</span> <span class="st">"BrayCurtis"</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://rdrr.io/pkg/scater/man/plotReducedDim.html">plotReducedDim</a></span><span class="op">(</span><span class="st">"BrayCurtis"</span>, colour_by <span class="op">=</span> <span class="st">"alcohol"</span>, shape_by <span class="op">=</span> <span class="st">"alcohol"</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="st">"PCo 1"</span>, y <span class="op">=</span> <span class="st">"PCo 2"</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu">guides</span><span class="op">(</span>colour <span class="op">=</span> <span class="fu">guide_legend</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Alcohol"</span><span class="op">)</span>, shape <span class="op">=</span> <span class="fu">guide_legend</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Alcohol"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu">theme</span><span class="op">(</span>legend.position <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0.90</span>, <span class="fl">0.85</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<div class="figure">
<img src="curatedMetagenomicData_files/figure-html/unnamed-chunk-20-1.png" alt="Beta Diversity - Bray–Curtis PCoA" width="700"><p class="caption">
Beta Diversity - Bray–Curtis PCoA
</p>
</div>
</div>
<div id="umap" class="section level3">
<h3 class="hasAnchor">
<a href="#umap" class="anchor"></a>UMAP</h3>
<p>To calculate the UMAP coordinates of every sample in our study we will use the <code><a href="https://rdrr.io/pkg/scater/man/runUMAP.html">runUMAP()</a></code> method from the <em><a href="https://bioconductor.org/packages/3.13/scater">scater</a></em> package package, as it handles the task in a single line.</p>
<p>To quickly plot the results of beta diversity analysis, the <code><a href="https://rdrr.io/pkg/scater/man/plotReducedDim.html">plotReducedDim()</a></code> method from the <em><a href="https://bioconductor.org/packages/3.13/scater">scater</a></em> package is used along with <em><a href="https://CRAN.R-project.org/package=ggplot2">ggplot2</a></em> syntax again.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">alcoholStudy</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://rdrr.io/pkg/scater/man/runUMAP.html">runUMAP</a></span><span class="op">(</span>exprs_values <span class="op">=</span> <span class="st">"relative_abundance"</span>, altexp <span class="op">=</span> <span class="st">"Genus"</span>, name <span class="op">=</span> <span class="st">"UMAP"</span><span class="op">)</span> <span class="op">%&gt;%</span>
    <span class="fu"><a href="https://rdrr.io/pkg/scater/man/plotReducedDim.html">plotReducedDim</a></span><span class="op">(</span><span class="st">"UMAP"</span>, colour_by <span class="op">=</span> <span class="st">"alcohol"</span>, shape_by <span class="op">=</span> <span class="st">"alcohol"</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="st">"UMAP 1"</span>, y <span class="op">=</span> <span class="st">"UMAP 2"</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu">guides</span><span class="op">(</span>colour <span class="op">=</span> <span class="fu">guide_legend</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Alcohol"</span><span class="op">)</span>, shape <span class="op">=</span> <span class="fu">guide_legend</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Alcohol"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu">theme</span><span class="op">(</span>legend.position <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0.90</span>, <span class="fl">0.85</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<div class="figure">
<img src="curatedMetagenomicData_files/figure-html/unnamed-chunk-21-1.png" alt="Beta Diversity - UMAP (Uniform Manifold Approximation and Projection)" width="700"><p class="caption">
Beta Diversity - UMAP (Uniform Manifold Approximation and Projection)
</p>
</div>
</div>
</div>
<div id="differential-abundance" class="section level2">
<h2 class="hasAnchor">
<a href="#differential-abundance" class="anchor"></a>Differential Abundance</h2>
<p>Next, it would be desirable to establish which microbes are differentially abundant between the two groups (those who consume alcohol, and those who do not). The <em><a href="https://bioconductor.org/packages/3.13/lefser">lefser</a></em> and <em><a href="https://bioconductor.org/packages/3.13/ANCOMBC">ANCOMBC</a></em> packages feature excellent methods to perform this tasks; however, code is not included here to avoid including many <code>Suggests</code> packages - <em><a href="https://bioconductor.org/packages/3.13/curatedMetagenomicData">curatedMetagenomicData</a></em> had far too many of these in the the previous version and is now very lean. There is a repository of analyses, <a href="https://github.com/waldronlab/curatedMetagenomicAnalyses">curatedMetagenomicAnalyses</a>, on GitHub and a forthcoming paper that will feature extensive demonstrations of analyses - but for now, the suggestions above will have to suffice.</p>
</div>
</div>
<div id="type-conversion" class="section level1">
<h1 class="hasAnchor">
<a href="#type-conversion" class="anchor"></a>Type Conversion</h1>
<p>Finally, the <em><a href="https://bioconductor.org/packages/3.13/curatedMetagenomicData">curatedMetagenomicData</a></em> package previously had methods for conversion to <code>phyloseq</code> class objects, and they have been removed in the latest release. It is likely that some users will still want to do analysis using <em><a href="https://bioconductor.org/packages/3.13/phyloseq">phyloseq</a></em>, and we would like to help them do so - it is just easier if we don’t have to maintain the conversion method ourselves. As such, the <em><a href="https://bioconductor.org/packages/3.13/mia">mia</a></em> package has a method, <code>makePhyloseqFromTreeSummarizedExperiment</code>, that will readily do the conversion - users needing this functionality are advised to use it.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/mia/man/makePhyloseqFromTreeSummarizedExperiment.html">makePhyloseqFromTreeSummarizedExperiment</a></span><span class="op">(</span><span class="va">alcoholStudy</span>, abund_values <span class="op">=</span> <span class="st">"relative_abundance"</span><span class="op">)</span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Lucas schiffer, Levi Waldron.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
